### PR TITLE
Add dataset field versioning support

### PR DESCRIPTION
**Pull request by wslulciuc**

### CODE REVIEW

File: src/main/java/marquez/db/DbTableVersionDao.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import marquez.db.mappers.DbTableVersionRowMapper;
+import marquez.db.models.DbTableVersionRow;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.sqlobject.SqlObject;
+import org.jdbi.v3.sqlobject.config.RegisterRowMapper;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+
+@RegisterRowMapper(DbTableVersionRowMapper.class)
+public interface DbTableVersionDao extends SqlObject {
+  default void insert(DbTableVersionRow row) {
```
Comment (fanrukun) [inline src/main/java/marquez/db/DbTableVersionDao.java]: I assume this insert function lives in a transaction context. correct ?

File: src/main/resources/db/migration/V4__add_db_table_versions.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+CREATE TABLE db_columns (
+  uuid         UUID PRIMARY KEY,
+  type         VARCHAR(64) NOT NULL,
+  created_at   TIMESTAMP NOT NULL,
+  updated_at   TIMESTAMP NOT NULL,
+  dataset_uuid UUID REFERENCES datasets(uuid),
+  name         UUID NOT NULL,
+  desscription UUID,
```
Comment (fanrukun) [inline src/main/resources/db/migration/V4__add_db_table_versions.sql]: typo ?

Comment (fanrukun) [reply src/main/resources/db/migration/V4__add_db_table_versions.sql]: alos should the type of name/description be UUID ?  shouldn't they be varchar ?

Comment (wslulciuc) [reply src/main/resources/db/migration/V4__add_db_table_versions.sql]: Great catch!

File: src/main/java/marquez/db/DbTableVersionDao.java (hunk 1 above)
Comment (wslulciuc) [reply src/main/java/marquez/db/DbTableVersionDao.java]: Yep! It definitely should 👍

**Pull request merged**